### PR TITLE
#761 Recovery page layout

### PR DESCRIPTION
**Pull request by sandrina-p**
Part of #761 

recovery-3

Implemented the layout with a dummy logic. The real logic will be done in a future PR by @taoeffect  

### Notes:
To see some specific scenario, the code needs to be modified.

- **App banner visibility:** Go to `frontend/main.js`, line 99. Change the key `isCorrupted` to true.

- **Troubleshooting initial status:** Go to `frontend/containers/user-settings/Troubleshooting.vue`. In line 80, change const `status` from `'ok'` to `'corrupted'`.
- **Recovery result:** Same file as before, line 124. Change the condition `false|true` to throw a dummy error and "break" the recovery process.

### CODE REVIEW

**Review by mmbotelho — CHANGES_REQUESTED** [review]
- Clicking on "re-sync your app data" should open on the same tab (current behavior is to open a new tab).

Other than that it's perfect!

**Comment by sandrina-p** [comment]
> Clicking on "re-sync your app data" should open on the same tab (current behavior is to open a new tab).

@mmbotelho, that's a current technical limitation. :/ The same happens when we click "Report the error" in a scoped error message.
We did that because if we "open on the same tab" it will actually refresh the page and lose the current page context. For example, if it happens when trying to fill a formulary.

However, in this scenario, (re-sync data) it may be okay to lose the current page context because the App is already broken anyway, so it's not that important.

TLDR: Changed. It now opens on the same tab :)

**Review by mmbotelho — DISMISSED** [review]
Approved, awesome work @sandrina-p !

File: frontend/main.js [hunk 1]
```diff
@@ -95,7 +95,8 @@ async function startApp () {
         ephemeral: {
           syncs: [],
           // TODO/REVIEW page can load with already loggedin. -> this.$store.state.loggedIn ? 'yes' : 'no'
-          finishedLogin: 'no'
+          finishedLogin: 'no',
+          isCorrupted: true // TODO #761
```
Comment (taoeffect) [inline frontend/main.js]: Can't merge while this is set to true; although having it set to true has helped me find a bug! 😄

File: frontend/main.js [hunk 2]
```diff
@@ -124,6 +125,16 @@ async function startApp () {
       // setTimeout(() => {
       //   this.$refs.bannerGeneral.show(this.L('Trying to reconnect...'), 'wifi')
       // }, 2500)
+
+      if (this.ephemeral.isCorrupted) {
+        this.$refs.bannerGeneral.danger(
+          this.L('Your app seems to be corrupted. Please {a_}re-sync your app data.{_a}', {
+            'a_': '<a class="link" href="/app/dashboard?modal=UserSettingsModal&section=troubleshooting">',
```
Comment (taoeffect) [inline frontend/main.js]: So... do we want to be able to access this page while we are not part of a group?

I think we might want to be able to do that... e.g. if the app is corrupted and therefore we can't access our group, it might make sense to be able to visit this as long as we're logged in.

That can be done either in this PR or in a future PR (in which case an issue is needed for that). Your choice!

**Review by taoeffect — CHANGES_REQUESTED** [review]
Looks good! Almost ready to merge, just a few things:

- `isCorrupted` needs to be set back to false
- See comment about accessing recovery page while outside of a group... two ways to proceed there, up to you!
- When I clicked Re-Sync the progress bar looked weird/broken in Firefox Dev Edition:

Screen Shot 2020-05-06 at 4 37 14 PM

File: frontend/main.js (hunk 1 above)
Comment (sandrina-p) [reply frontend/main.js]: Changed to false!

File: frontend/main.js (hunk 2 above)
Comment (sandrina-p) [reply frontend/main.js]: Hum... nice question. I made it work on the last commit. Please take a look.

Comment (taoeffect) [reply frontend/main.js]: I think the file `router.js` needs to be updated... it just redirects back to welcome page to create a group. You can add a TODO for that or implement in this PR, your choice!

Comment (sandrina-p) [reply frontend/main.js]: Why do we need to update `router.js` exactly? I tested and it worked - it opens the recovery page.

Note that this is dumb: The URL is created on mount, so if we change pages, the URL is not updated (the pathname), and that's maybe why you are facing problems. This was just a dummy implementation of the banner link. IRL the banner would appear at the right moment, and there the URL would be correct.

Comment (taoeffect) [reply frontend/main.js]: > Why do we need to update router.js exactly? I tested and it worked - it opens the recovery page.

... weird. Doesn't work for me 🤷‍♂️ 

I copy/pasted the URL and manually loaded it. Was still redirected to the welcome page where it asks you to create a group. I'm guessing it's one of the guards that's causing it. Did you try in Firefox?

Comment (sandrina-p) [reply frontend/main.js]: When you open a page with no group and click the recovery link,
it should start with `/app/?modal=`.

When you open a tab with group, and click the recovery link,
it should start with `/app/dashboard/modal=`

Comment (taoeffect) [reply frontend/main.js]: Ah! You're right I must've been copying the wrong link! Will approve now!

**Review by taoeffect — APPROVED** [review]
Fantastic! 🎉
